sqlglot/transforms.py

```diff
@@ -187,7 +187,7 @@ def new_name(names: t.Set[str], name: str) -> str:
                 table=[series_alias],
             )
 
-            for select in list(expression.selects):
+            for select in expression.selects:
```

Comment:
i added this on purpose due to an infinite loop, is this not needed anymore?

Reply:
expression.expressions is getting mutated  on line 230

Reply:
Damn thanks, I changed it on a whim and didn't notice the mutation happening. Reverted.